packages/block-library/src/paragraph/edit.js

```diff
@@ -108,28 +109,31 @@ function ParagraphBlock( {
 		} ),
 		style: { direction },
 	} );
+	const blockEditingMode = useBlockEditingMode();
```

Comment:
This is adding a store subscription for every paragraph block and is thus causing a performance regression.

Reply:
Thanks for catching it. There's a few blocks that use `useBlockEditingMode` like this, so that's why I didn't think it would be much of an issue, but I should have checked the performance results.